internal/plugin/installer/plugin_structure.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright The Helm Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package installer
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"helm.sh/helm/v4/internal/plugin"
+)
+
+// detectPluginRoot searches for plugin.yaml in the extracted directory
+// and returns the path to the directory containing it.
+// This handles cases where the tarball contains the plugin in a subdirectory.
+func detectPluginRoot(extractDir string) (string, error) {
```

Comment:
I'm a little confused by the use-case for `detectPluginRoot` checking one directory deep for a `plugin.yaml` file. Might just be a convenience function.

What it makes me think of is how bitnami charts do something like have a `charts/` directory, maybe someone would have plugins in a `plugins/` directory. But if I ran `helm plugin install`, on `plugins/` then it would be the first directory with a `plugin.yaml` file in it.

Another kind of user might expect a `helm plugin install` to install ALL plugins in that directory.